packages/story-editor/src/components/panels/design/textAccessibility/textAccessibility.js

```diff
@@ -0,0 +1,167 @@
+} from '@googleforcreators/design-system';
+import { getTextElementTagNames } from '@googleforcreators/output';
+import { useFeature } from 'flagged';
+import { useState, useEffect } from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import { Row } from '../../../form';
+import { SimplePanel } from '../../panel';
+import { useHighlights, states, styles } from '../../../../app/highlights';
+import { MULTIPLE_VALUE, MULTIPLE_DISPLAY_VALUE } from '../../../../constants';
+import { combineElementsWithTags } from './utils';
+
+// Options to be mapped for the dropdown
+const semanticHeadingsOptionsMap = [
+  {
+    label: __('Heading 1', 'web-stories'),
+    value: 'h1',
+  },
+  {
+    label: __('Heading 2', 'web-stories'),
+    value: 'h2',
+  },
+  {
+    label: __('Heading 3', 'web-stories'),
+    value: 'h3',
+  },
+  {
+    label: __('Paragraph', 'web-stories'),
+    value: 'p',
+  },
+  {
+    label: __('Automatic', 'web-stories'),
+    value: 'auto',
+  },
+];
+
+function TextAccessibilityPanel({ selectedElements, pushUpdate }) {
+  // Feature flagging for Semantic Headings
+  const showSemanticHeadings = useFeature('showSemanticHeadings');
+  const [selectedTextElements, setSelectedTextElements] = useState([]);
+  const [currentTags, setCurrentTags] = useState([]);
+  useEffect(() => {
+    // We only want to impact elements if all selected elements are of type text
+    const textElements = selectedElements.filter(({ type }) => 'text' === type);
+    // Then we want to get the text tags for the elements if not already defined
+    const textTags = getTextElementTagNames(textElements);
+    // Then combine the elements with their associated tag
+    const newElements = combineElementsWithTags(textElements, textTags);
+    // And then set them into component state
+    setSelectedTextElements(newElements);
+    setCurrentTags(Array.from(textTags.values()));
+  }, [selectedElements, setSelectedTextElements]);
+  const { highlight, resetHighlight, cancelHighlight } = useHighlights(
+    (state) => ({
+      highlight: state[states.ASSISTIVE_TEXT],
+      resetHighlight: state.onFocusOut,
+      cancelHighlight: state.cancelEffect,
+    })
+  );
+  if (!showSemanticHeadings) {
+    return null;
+  }
+  // Map all types of tag names in the selected elements
+  // and then convert to an Array for usage
+  const handleChange = (ev, value) => {
+    selectedElements.map(() => {
+      return pushUpdate({ tagName: value });
+    });
+    // Update the selected text elements with the
+    // value chosen in the text accessibility dropdown
+    const newTags = getTextElementTagNames(selectedTextElements, value);
+    setCurrentTags(Array.from(newTags.values()));
+  };
+  // Check if tags match for selected item in dropdown
+  // If they don't match, we want to show 'Mixed'
+  // However, if they match, show the matching value
+  const currentValue =
+    new Set(currentTags || []).size === 1 ? currentTags[0] : '((MULTIPLE))';
```

Comment:
Let's use the available const for that.

```suggestion
    new Set(currentTags || []).size === 1 ? currentTags[0] : MULTIPLE_VALUE;
```

Reply:
Will do!

Reply:
Was changed as part of merging in https://github.com/GoogleForCreators/web-stories-wp/pull/11888/commits/6dc71c5d492413303d5c952fe5ccfef660dcb476